integration-tests/src/test/burn.test.ts

```diff
@@ -0,0 +1,117 @@
+import { Blockchain, KeyringPair, creditcoinApi, BN } from 'creditcoin-js';
+import { forElapsedBlocks, testData } from 'creditcoin-js/lib/testUtils';
+import { CreditcoinApi } from 'creditcoin-js/lib/types';
+import { mnemonicGenerate } from '@polkadot/util-crypto';
+import { extractFee, describeIf } from '../utils';
+
+describeIf((global as any).CREDITCOIN_EXECUTE_SETUP_AUTHORITY, 'burn', () => {
+    let ccApi: CreditcoinApi;
+    let sudoSigner: KeyringPair;
+    let wallet: KeyringPair;
+
+    const testingData = testData(
+        (global as any).CREDITCOIN_ETHEREUM_CHAIN as Blockchain,
+        (global as any).CREDITCOIN_CREATE_WALLET,
+    );
+    const { keyring } = testingData;
+
+    beforeAll(async () => {
+        ccApi = await creditcoinApi((global as any).CREDITCOIN_API_URL);
+        sudoSigner = (global as any).CREDITCOIN_CREATE_SIGNER(keyring, 'lender');
+        wallet = (global as any).CREDITCOIN_CREATE_SIGNER(keyring, 'borrower');
+    });
+
+    afterAll(async () => await ccApi.api.disconnect());
+
+    it('burn_all works as expected', async (): Promise<void> => {
+        const {
+            api,
+            extrinsics: {},
+        } = ccApi;
+
+        await api.tx.sudo
+            .sudo(api.tx.balances.setBalance(wallet.address, new BN('1000000000000000000', 10), 0))
+            .signAndSend(sudoSigner, { nonce: -1 });
+        await forElapsedBlocks(api);
+
+        const starting = await api.derive.balances.all(wallet.address);
+        expect(starting.freeBalance.isZero()).toBe(false);
+
+        await api.tx.creditcoin.burnAll(sudoSigner.address).signAndSend(wallet);
+        await forElapsedBlocks(api);
+
+        const ending = await api.derive.balances.all(wallet.address);
+        expect(ending.freeBalance.isZero()).toBe(true);
+    }, 100_000);
+
+    it('burn works as expected', async (): Promise<void> => {
+        const burner = keyring.addFromMnemonic(mnemonicGenerate(12));
+
+        const {
+            api,
+            extrinsics: {},
+        } = ccApi;
+
+        await api.tx.sudo
+            .sudo(api.tx.balances.setBalance(burner.address, new BN('1000000000000000000', 10), 0))
```

Comment:
If I am not mistaken this value is 1 CTC specified in CREDO. Let's use something like this for readability:

```
const oneCtc = (1 * CREDO_PER_CTC).toString();
```

See also the commit log at https://github.com/gluwa/creditcoin/commit/a8314dff54272897e6f75a0423d9a6881ef805d7 for some extra context.